### PR TITLE
use KVM specific k8scloudconfig version

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -8,10 +8,8 @@ import:
   version: master
 - package: github.com/giantswarm/kvmtpr
   version: master
-# Pinned to a branch with w/a for https://github.com/giantswarm/kvm-operator/pull/216.
-# Make sure to rebase k8scloudconfig "kvm-operator-disable-rbac" with "master"
 - package: github.com/giantswarm/k8scloudconfig
-  version: kvm-operator-disable-rbac
+  version: master
```
Comment (xh3b4sd) [inline glide.yaml]: There is no weird vendor hack anymore. We version now within the code.

File: service/cloudconfigv1/cloud_config.go [hunk 2]
```diff
@@ -57,13 +57,17 @@ func (c *CloudConfig) NewMasterTemplate(customObject kvmtpr.CustomObject, certs
 
 	// TODO remove defaulting as soon as custom objects are configured.
 	if customObject.Spec.Cluster.Version == "" {
-		customObject.Spec.Cluster.Version = string(cloudconfig.V_0_1_0)
+		customObject.Spec.Cluster.Version = string(cloudconfig.V_1_0_0)
+	}
+	// TODO remove defaulting as soon as custom objects are migrated.
+	if customObject.Spec.Cluster.Version == string(cloudconfig.V_0_1_0) {
+		customObject.Spec.Cluster.Version = string(cloudconfig.V_1_0_0)
```
Comment (xh3b4sd) [inline service/cloudconfigv1/cloud_config.go]: We use [`1.0.0`](https://github.com/giantswarm/k8scloudconfig/blob/ea70bc4ea6fc4c3214c0a71e3c6a95ba3be5fd32/template.go#L16-L18) now which is the same as `0.1.0` but with experimental encryption disabled. 

The weird conditionals here have to stay a bit until we reconcile on the cluster-service/kubernetesd side. There is https://github.com/giantswarm/giantswarm/issues/2148 to track this.

File: service/cloudconfigv1/cloud_config.go [hunk 3]
```diff
@@ -57,13 +57,17 @@ func (c *CloudConfig) NewMasterTemplate(customObject kvmtpr.CustomObject, certs
...
 	}
 
 	var template string
 
 	switch customObject.Spec.Cluster.Version {
-	case string(cloudconfig.V_0_1_0):
+	case string(cloudconfig.V_1_0_0):
 		template, err = v_0_1_0MasterTemplate(customObject, certs, node)
```
Comment (xh3b4sd) [inline service/cloudconfigv1/cloud_config.go]: The naming here is weird. I will provide a PR to fix this separately.

File: service/cloudconfigv1/cloud_config.go (hunk 2 above)
Comment (rossf7) [reply service/cloudconfigv1/cloud_config.go]: Ack. We'll need to do the same in `aws-operator`.

Comment (xh3b4sd) [reply service/cloudconfigv1/cloud_config.go]: To be on the same page, in AWS you do not need to do anything now. What we need to make sure is that we use `2.0.0` when migrating to CRDs.

Comment (rossf7) [reply service/cloudconfigv1/cloud_config.go]: OK, that makes sense. Thanks for clarifying.

File: service/cloudconfigv1/v_1_0_0_master_template.go [hunk 4]
```diff
@@ -3,18 +3,18 @@ package cloudconfigv1
 import (
 	"github.com/giantswarm/certificatetpr"
 	clustertprspec "github.com/giantswarm/clustertpr/spec"
-	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_0_1_0"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_1_0_0"
```
Comment (xh3b4sd) [inline service/cloudconfigv1/v_1_0_0_master_template.go]: I had to fix the import.

File: service/cloudconfigv1/v_1_0_0_master_template.go [hunk 5]
```diff
@@ -3,18 +3,18 @@ package cloudconfigv1
...
 	"github.com/giantswarm/kvmtpr"
 	"github.com/giantswarm/microerror"
 )
 
-func v_0_1_0MasterTemplate(customObject kvmtpr.CustomObject, certs certificatetpr.AssetsBundle, node clustertprspec.Node) (string, error) {
+func v_1_0_0MasterTemplate(customObject kvmtpr.CustomObject, certs certificatetpr.AssetsBundle, node clustertprspec.Node) (string, error) {
```
Comment (xh3b4sd) [inline service/cloudconfigv1/v_1_0_0_master_template.go]: I also did the renaming now because the naming reflects the import and the supported version.

**Comment by xh3b4sd** [comment]
Going with this now.

**Pull request merged**